<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Appendix A: Loading Expression Data in R | Analysis of Gene Expression</title>
  <meta name="description" content="Course Syllabus for BFV: Theme 07 course" />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="A Appendix A: Loading Expression Data in R | Analysis of Gene Expression" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course Syllabus for BFV: Theme 07 course" />
  <meta name="github-repo" content="mkempenaar/gene_expression_analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Appendix A: Loading Expression Data in R | Analysis of Gene Expression" />
  
  <meta name="twitter:description" content="Course Syllabus for BFV: Theme 07 course" />
  

<meta name="author" content="Marcel Kempenaar" />


<meta name="date" content="2022-04-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter-5.html"/>
<link rel="next" href="a2-annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysis of Gene Expression</a></li>

<li class="divider"></li>
<li><a href="index.html#preface">Preface<span></span></a>
<ul>
<li><a href="index.html#analysis-of-gene-expression-project">Analysis of Gene Expression Project<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chapter-1.html"><a href="chapter-1.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="chapter-1.html"><a href="chapter-1.html#capstone-project---analysis-of-gene-expression"><i class="fa fa-check"></i><b>1.1</b> Capstone Project - Analysis of Gene Expression<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="chapter-1.html"><a href="chapter-1.html#project-deliverables"><i class="fa fa-check"></i><b>1.2</b> Project Deliverables<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="chapter-1.html"><a href="chapter-1.html#project-schedule"><i class="fa fa-check"></i><b>1.3</b> Project Schedule<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="chapter-1.html"><a href="chapter-1.html#grading"><i class="fa fa-check"></i><b>1.4</b> Grading<span></span></a></li>
<li class="chapter" data-level="1.5" data-path="chapter-1.html"><a href="chapter-1.html#lab-journal"><i class="fa fa-check"></i><b>1.5</b> <em>Lab Journal</em><span></span></a></li>
<li class="chapter" data-level="1.6" data-path="chapter-1.html"><a href="chapter-1.html#article"><i class="fa fa-check"></i><b>1.6</b> Article<span></span></a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="chapter-1.html"><a href="chapter-1.html#installing-the-article-template"><i class="fa fa-check"></i><b>1.6.1</b> Installing the Article Template<span></span></a></li>
<li class="chapter" data-level="1.6.2" data-path="chapter-1.html"><a href="chapter-1.html#using-the-template"><i class="fa fa-check"></i><b>1.6.2</b> Using the Template<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter-2.html"><a href="chapter-2.html"><i class="fa fa-check"></i><b>2</b> Discovering Data Sets<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="chapter-2.html"><a href="chapter-2.html#finding-public-data"><i class="fa fa-check"></i><b>2.1</b> Finding Public Data Sets of Interest<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="chapter-2.html"><a href="chapter-2.html#ncbi-geo"><i class="fa fa-check"></i><b>2.2</b> The NCBI <em>Gene Expression Omnibus</em><span></span></a></li>
<li class="chapter" data-level="2.3" data-path="chapter-2.html"><a href="chapter-2.html#project-proposal_ch2"><i class="fa fa-check"></i><b>2.3</b> Creating a Project Proposal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter-3.html"><a href="chapter-3.html"><i class="fa fa-check"></i><b>3</b> Exploratory Data Analysis<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="chapter-3.html"><a href="chapter-3.html#loading-data"><i class="fa fa-check"></i><b>3.1</b> Loading data into R<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="chapter-3.html"><a href="chapter-3.html#ch3_example_data"><i class="fa fa-check"></i><b>3.2</b> Example Data<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="chapter-3.html"><a href="chapter-3.html#EDA_part1"><i class="fa fa-check"></i><b>3.3</b> Visualizing using <code>boxplot</code> and <code>density plot</code><span></span></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="chapter-3.html"><a href="chapter-3.html#eda_statistics"><i class="fa fa-check"></i><b>3.3.1</b> Statistics<span></span></a></li>
<li class="chapter" data-level="3.3.2" data-path="chapter-3.html"><a href="chapter-3.html#eda_boxplots"><i class="fa fa-check"></i><b>3.3.2</b> Boxplots<span></span></a></li>
<li class="chapter" data-level="3.3.3" data-path="chapter-3.html"><a href="chapter-3.html#eda_density_plot"><i class="fa fa-check"></i><b>3.3.3</b> Density Plots<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="chapter-3.html"><a href="chapter-3.html#EDA_part2"><i class="fa fa-check"></i><b>3.4</b> Visualizing using <code>heatmap</code> and <code>MDS</code><span></span></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="chapter-3.html"><a href="chapter-3.html#normalization"><i class="fa fa-check"></i><b>3.4.1</b> Normalization<span></span></a></li>
<li class="chapter" data-level="3.4.2" data-path="chapter-3.html"><a href="chapter-3.html#eda_distance"><i class="fa fa-check"></i><b>3.4.2</b> Distance Calculation<span></span></a></li>
<li class="chapter" data-level="3.4.3" data-path="chapter-3.html"><a href="chapter-3.html#eda_heatmap"><i class="fa fa-check"></i><b>3.4.3</b> Sample Distances using a Heatmap<span></span></a></li>
<li class="chapter" data-level="3.4.4" data-path="chapter-3.html"><a href="chapter-3.html#eda_mds"><i class="fa fa-check"></i><b>3.4.4</b> Multi-Dimensional Scaling<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="chapter-3.html"><a href="chapter-3.html#cleaning-data"><i class="fa fa-check"></i><b>3.5</b> Cleaning Data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter-4.html"><a href="chapter-4.html"><i class="fa fa-check"></i><b>4</b> Discovering Differentialy Expressed Genes (DEGs)<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="chapter-4.html"><a href="chapter-4.html#deg-preprocessing"><i class="fa fa-check"></i><b>4.1</b> Pre-processing<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="chapter-4.html"><a href="chapter-4.html#deg-fc"><i class="fa fa-check"></i><b>4.2</b> The <em>Fold Change</em> Value<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="chapter-4.html"><a href="chapter-4.html#deg-bioconductor"><i class="fa fa-check"></i><b>4.3</b> Using Bioconductor Packages<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="chapter-4.html"><a href="chapter-4.html#deg-design"><i class="fa fa-check"></i><b>4.3.1</b> The <code>Design</code> (matrix)<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter-4.html"><a href="chapter-4.html#deg-deseq2"><i class="fa fa-check"></i><b>4.3.2</b> <code>DESeq2</code><span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter-4.html"><a href="chapter-4.html#deg-edger"><i class="fa fa-check"></i><b>4.3.3</b> <code>edgeR</code><span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter-5.html"><a href="chapter-5.html"><i class="fa fa-check"></i><b>5</b> Data Analysis and Visualization<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="chapter-5.html"><a href="chapter-5.html#volcano"><i class="fa fa-check"></i><b>5.1</b> Volcano Plot<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="chapter-5.html"><a href="chapter-5.html#venn-diagram"><i class="fa fa-check"></i><b>5.2</b> Venn Diagram<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="chapter-5.html"><a href="chapter-5.html#clustering"><i class="fa fa-check"></i><b>5.3</b> Clustering<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="chapter-5.html"><a href="chapter-5.html#pathway-analysis"><i class="fa fa-check"></i><b>5.4</b> Pathway Analysis<span></span></a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="chapter-5.html"><a href="chapter-5.html#gene-functional-classification-using-david-100-degs"><i class="fa fa-check"></i><b>5.4.1</b> Gene Functional Classification using DAVID (&gt;100 DEGs)<span></span></a></li>
<li class="chapter" data-level="5.4.2" data-path="chapter-5.html"><a href="chapter-5.html#kegg-pathway-visualization-10-degs"><i class="fa fa-check"></i><b>5.4.2</b> KEGG-pathway Visualization (&gt;10 DEGs)<span></span></a></li>
<li class="chapter" data-level="5.4.3" data-path="chapter-5.html"><a href="chapter-5.html#spia"><i class="fa fa-check"></i><b>5.4.3</b> Signaling Pathway Impact Analysis (&gt;20 DEGs)<span></span></a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix<span></span></b></span></li>
<li class="chapter" data-level="A" data-path="a1-data_loading.html"><a href="a1-data_loading.html"><i class="fa fa-check"></i><b>A</b> Appendix A: Loading Expression Data in R<span></span></a>
<ul>
<li class="chapter" data-level="A.1" data-path="a1-data_loading.html"><a href="a1-data_loading.html#expl-data"><i class="fa fa-check"></i><b>A.1</b> Exploring the Available Project Data<span></span></a></li>
<li class="chapter" data-level="A.2" data-path="a1-data_loading.html"><a href="a1-data_loading.html#a1-batch_data_loading"><i class="fa fa-check"></i><b>A.2</b> Batch Loading Expression Data<span></span></a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="a1-data_loading.html"><a href="a1-data_loading.html#decompressing"><i class="fa fa-check"></i><b>A.2.1</b> Decompressing<span></span></a></li>
<li class="chapter" data-level="A.2.2" data-path="a1-data_loading.html"><a href="a1-data_loading.html#determining-data-format"><i class="fa fa-check"></i><b>A.2.2</b> Determining Data Format<span></span></a></li>
<li class="chapter" data-level="A.2.3" data-path="a1-data_loading.html"><a href="a1-data_loading.html#loading-data-1"><i class="fa fa-check"></i><b>A.2.3</b> Loading Data<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="a2-annotation.html"><a href="a2-annotation.html"><i class="fa fa-check"></i><b>B</b> Annotating an RNA-Seq Experiment<span></span></a>
<ul>
<li class="chapter" data-level="B.1" data-path="a2-annotation.html"><a href="a2-annotation.html#download-geo"><i class="fa fa-check"></i><b>B.1</b> Downloading annotation data from GEO<span></span></a></li>
<li class="chapter" data-level="B.2" data-path="a2-annotation.html"><a href="a2-annotation.html#manually-annotate"><i class="fa fa-check"></i><b>B.2</b> Manual Data Annotation<span></span></a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="a2-annotation.html"><a href="a2-annotation.html#annotationDBI"><i class="fa fa-check"></i><b>B.2.1</b> Using <code>AnnotationDBI</code><span></span></a></li>
<li class="chapter" data-level="B.2.2" data-path="a2-annotation.html"><a href="a2-annotation.html#using-biomart"><i class="fa fa-check"></i><b>B.2.2</b> Using <code>biomaRt</code><span></span></a></li>
</ul></li>
<li class="chapter" data-level="B.3" data-path="a2-annotation.html"><a href="a2-annotation.html#references"><i class="fa fa-check"></i><b>B.3</b> References<span></span></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of Gene Expression</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="a1-data_loading" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">A</span> Appendix A: Loading Expression Data in R<a href="a1-data_loading.html#a1-data_loading" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="expl-data" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">A.1</span> Exploring the Available Project Data<a href="a1-data_loading.html#expl-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Whether you’ve found a dataset through the SRA or GEO, we want to get the data into R to start working with it. For now, we will only download the <strong>count data</strong> which is most likely stored in a <code>TXT</code> or <code>XLS(X)</code> file format:</p>
<p><strong>Data formats</strong>:</p>
<ul>
<li><strong>TXT</strong>: simple text file containing a minimum of two columns (either tab or comma separated containing i.e. gene / transcript identifier and one of the above mentioned data types).
<ul>
<li>However it can also contain up to 10 data columns either including more information regarding the gene/ transcript (i.e. gene ID, name, symbol, chromosome, start/ stop position, length, etc.) or more numerical columns (i.e. raw read count, normalized readcount, FPKM, etc.).</li>
</ul></li>
<li><strong>XLS(X)</strong>: Microsoft Excel file containing the same data columns as mentioned in the TXT files.</li>
</ul>
<p><strong>Data types</strong>:</p>
<ul>
<li><strong>Read Count</strong> (<strong>required</strong>; simple <em>raw</em> count of mapped reads to a certain gene or transcript).</li>
<li><strong>FPKM</strong> (Fragments Per Kilobase Million, <strong>F</strong> for <em>Fragment</em>, paired-end sequencing),</li>
<li><strong>RPKM</strong> (Reads Per Kilobase Million, <strong>R</strong> for <em>Read</em>, single-end sequencing),</li>
<li><strong>TPM</strong> (Transcripts Per Kilobase Million, <strong>T</strong> for <em>Transcript</em>, technique independent),</li>
</ul>
<p><strong>Layouts</strong></p>
<ul>
<li>Either one or more files <strong>per sample</strong> with one of the above data types or</li>
<li>One file containing the data types for <strong>all samples</strong> (with the samples as data columns in the file)</li>
</ul>
<p><strong>Please watch</strong> the video and read the page found at the <a href="http://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/">RNA-Seq blog</a> regarding the meaning and calculation of the above mentioned expression data formats or a more technical document found at <a href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">The farrago blog page</a>.</p>
<p>On GEO you can see what data <em>might</em> be availalbe in the <code>Supplementary</code> column, as shown below:</p>
<div class="figure"><span style="display:block;" id="fig:GEODataOverview"></span>
<img src="chapters/images/geo_data-overview.png" alt="Finding an experiment on GEO with a TXT file as supplementary data" width="1114" />
<p class="caption">
Figure A.1: Finding an experiment on GEO with a TXT file as supplementary data
</p>
</div>
<p>This overview on GEO contains many links which are <em>not</em> direct links to the items for that dataset, but can be used as <em>filter</em> for browsing the results. If you want to actually download the data, click on the <strong>GSE</strong> identifier (first column) which brings you to an overview for this experiment containing a lot of information about the experiment (subject, research institute, publication, design, etc.) and links to each individual sample (<strong>GSM</strong> identifier). Following the link to a sample shows information on how this sample was retrieved with often many (lab) protocols used. Sometimes there is a segment regarding “<em>Data Processing</em>” that refers to techniques and software used for the full analysis and might contain something like:</p>
<blockquote>
<p>… Differential expression testing between sample groups was performed in Rstudio (v. 1.0.36) using DESeq2 (v.1.14.1) …</p>
</blockquote>
<p>Back on the experiment overview page you’ll see a (variable) number of links to data files belonging to this experiment, see <span class="math inline">\(\ref{fig:GEO_Data-download}\)</span>. For now we are only interested in the count-data which is stored in the <code>TXT</code>-file (see the column <code>File type/resource</code>) called <code>GSE97406_RAW.tar</code>. This file contains <em>all</em> the data that we need, even though it is only <strong>220Kb</strong> in size where the experiment started with about <strong>5Gb</strong> of read-data for a small bacteria:</p>
<div class="figure"><span style="display:block;" id="fig:GEODataDownload"></span>
<img src="chapters/images/geo_data-download.png" alt="Finding the supplementary data in a GEO record" width="697" />
<p class="caption">
Figure A.2: Finding the supplementary data in a GEO record
</p>
</div>
<p>As a bioinformatician we love to compress all the files so for this particular example, we download the <code>tar</code>-archive file, extract it to find a folder with another archive file for each sample. After extracting these files we end up with 12 <code>TXT</code>-files with just two columns; a <em>gene</em> identifier and a raw <em>count</em> value (~4500 rows of data per sample):</p>
<table>
<caption><span id="tab:foo">Table A.1: </span> Contents of a <code>TXT</code>-file for sample <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2563998"><code>GSM2563998</code></a></caption>
<thead>
<tr class="header">
<th>Gene ID</th>
<th>Count Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>aaaD</td>
<td>0</td>
</tr>
<tr class="even">
<td>aaaE</td>
<td>0</td>
</tr>
<tr class="odd">
<td>aaeR</td>
<td>50</td>
</tr>
<tr class="even">
<td>aaeX</td>
<td>0</td>
</tr>
<tr class="odd">
<td>aas</td>
<td>118</td>
</tr>
<tr class="even">
<td>abgB</td>
<td>21</td>
</tr>
<tr class="odd">
<td>abgR</td>
<td>56</td>
</tr>
<tr class="even">
<td>abgT</td>
<td>0</td>
</tr>
<tr class="odd">
<td>abrB</td>
<td>11</td>
</tr>
<tr class="even">
<td>accA</td>
<td>453</td>
</tr>
<tr class="odd">
<td>accB</td>
<td>2492</td>
</tr>
<tr class="even">
<td>accC</td>
<td>1197</td>
</tr>
</tbody>
</table>
<p>Other experiments combine their samples in a single file where each column represents a sample. If you do get an experiment with one file per sample, we can programmatically (yes, even in R) load this data in batch.</p>
</div>
<div id="a1-batch_data_loading" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">A.2</span> Batch Loading Expression Data<a href="a1-data_loading.html#a1-batch_data_loading" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This code example shows how to batch-load multiple files containing expression (count) data for a <em>single</em> sample. The data for this example can be found on GEO with ID <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE109798">GSE109798</a>. Downloading the data for this experiment from GEO gives us a single <code>.tar</code> file called <code>GSE109798_RAW.tar</code>. Extracting this archive file nets us a folder with the following files:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="a1-data_loading.html#cb40-1" aria-hidden="true" tabindex="-1"></a>file.names <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&#39;./data/GSE109798_RAW/&#39;</span>)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Files</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GSM2970149_4T1E274.isoforms.results.txt</td>
</tr>
<tr class="even">
<td align="left">GSM2970150_4T1E266.isoforms.results.txt</td>
</tr>
<tr class="odd">
<td align="left">GSM2970151_4T1E247D.isoforms.results.txt</td>
</tr>
<tr class="even">
<td align="left">GSM2970152_4T1P2247A.isoforms.results.txt</td>
</tr>
<tr class="odd">
<td align="left">GSM2970153_4T1P2247G.isoforms.results.txt</td>
</tr>
<tr class="even">
<td align="left">GSM2970154_4T1P2247F.isoforms.results.txt</td>
</tr>
<tr class="odd">
<td align="left">GSM2970155_HCC1806E224B.isoforms.results.txt</td>
</tr>
<tr class="even">
<td align="left">GSM2970156_HCC1806E224A.isoforms.results.txt</td>
</tr>
<tr class="odd">
<td align="left">GSM2970157_HCC1806E224C.isoforms.results.txt</td>
</tr>
<tr class="even">
<td align="left">GSM2970158_HCC1806P2232A.isoforms.results.txt</td>
</tr>
<tr class="odd">
<td align="left">GSM2970159_HCC1806P2232B.isoforms.results.txt</td>
</tr>
<tr class="even">
<td align="left">GSM2970160_HCC1806P2230.isoforms.results.txt</td>
</tr>
</tbody>
</table>
<div id="decompressing" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">A.2.1</span> Decompressing<a href="a1-data_loading.html#decompressing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The file extension of all these files is <code>.txt.gz</code> which means that all files are compressed using <code>gzip</code> and need to be unpacked before they can be loaded. The easiest method is using the system <code>gunzip</code> command on all files which can be done from within R by <em>applying</em> the <code>gunzip</code> command using the <code>system</code> function on each file.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="a1-data_loading.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Change directory to where the files are stored</span></span>
<span id="cb41-2"><a href="a1-data_loading.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&#39;./data/GSE109798_RAW/&#39;</span>)</span>
<span id="cb41-3"><a href="a1-data_loading.html#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(file.names, <span class="at">FUN =</span> <span class="cf">function</span>(file.name) {</span>
<span id="cb41-4"><a href="a1-data_loading.html#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system</span>(<span class="fu">paste</span>(<span class="st">&quot;gunzip&quot;</span>, file.name)) </span>
<span id="cb41-5"><a href="a1-data_loading.html#cb41-5" aria-hidden="true" tabindex="-1"></a>  })</span></code></pre></div>
<p>Now we can update the <code>file.names</code> variable since each file name has changed.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="a1-data_loading.html#cb42-1" aria-hidden="true" tabindex="-1"></a>file.names <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&#39;./data/GSE109798_RAW/&#39;</span>)</span></code></pre></div>
</div>
<div id="determining-data-format" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">A.2.2</span> Determining Data Format<a href="a1-data_loading.html#determining-data-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we can inspect what the contents are of these files, assuming that they all have the same layout/ column names etc. to decide what we need to use for our analysis.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="a1-data_loading.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Call the system &#39;head&#39; tool to &#39;peek&#39; inside the file</span></span>
<span id="cb43-2"><a href="a1-data_loading.html#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="fu">paste0</span>(<span class="st">&quot;head &quot;</span>, <span class="st">&quot;./data/GSE109798_RAW/&quot;</span>, file.names[<span class="dv">1</span>]))</span></code></pre></div>
<table>
<colgroup>
<col width="17%" />
<col width="10%" />
<col width="8%" />
<col width="21%" />
<col width="19%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">transcript_id</th>
<th align="left">gene_id</th>
<th align="left">length</th>
<th align="left">effective_length</th>
<th align="left">expected_count</th>
<th align="left">TPM</th>
<th align="left">FPKM</th>
<th align="left">IsoPct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">uc007aet.1</td>
<td align="left">1</td>
<td align="left">3608</td>
<td align="left">3608.00</td>
<td align="left">1.82</td>
<td align="left">0.44</td>
<td align="left">0.24</td>
<td align="left">100.00</td>
</tr>
<tr class="even">
<td align="left">uc007aeu.1</td>
<td align="left">1</td>
<td align="left">3634</td>
<td align="left">3634.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
</tr>
<tr class="odd">
<td align="left">uc011whv.1</td>
<td align="left">10</td>
<td align="left">26</td>
<td align="left">26.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
</tr>
<tr class="even">
<td align="left">uc007amd.1</td>
<td align="left">100</td>
<td align="left">1823</td>
<td align="left">1823.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
</tr>
<tr class="odd">
<td align="left">uc007ame.1</td>
<td align="left">100</td>
<td align="left">4355</td>
<td align="left">4355.00</td>
<td align="left">1.32</td>
<td align="left">0.26</td>
<td align="left">0.15</td>
<td align="left">100.00</td>
</tr>
<tr class="even">
<td align="left">uc007dac.1</td>
<td align="left">1000</td>
<td align="left">1403</td>
<td align="left">1403.00</td>
<td align="left">2.00</td>
<td align="left">1.25</td>
<td align="left">0.70</td>
<td align="left">100.00</td>
</tr>
<tr class="odd">
<td align="left">uc008ajp.1</td>
<td align="left">10000</td>
<td align="left">1078</td>
<td align="left">1078.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
</tr>
<tr class="even">
<td align="left">uc012ajs.1</td>
<td align="left">10000</td>
<td align="left">1753</td>
<td align="left">1753.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
</tr>
<tr class="odd">
<td align="left">uc008ajq.1</td>
<td align="left">10001</td>
<td align="left">2046</td>
<td align="left">2046.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
<td align="left">0.00</td>
</tr>
</tbody>
</table>
<p>These files contain (much) more then just a count value for each gene as we can see columns such as (transcript) <em>length</em>, <em>TPM</em>, <em>FPKM</em>, etc. Also, the count-column is called <em>expected_count</em> which raises a few questions as well.</p>
<div class="rmdtip">
<p>
The <em>expected count</em> value is usable as it contains more information - compared to the raw count - then we actually require. The <em>expected</em> part results from <em>multimapped reads</em> where a single read mapped to multiple positions in the genome. As each transcript originates only from one location, this multimapped read is usually discarded. With the <em>expected count</em> though, instead of discarding the read completely it is <em>estimated</em> where it originates from and this is added as a <em>fraction</em> to the count value. So the value of <code>1.32</code> that we see on line 5 in the example above means an true count of <code>1</code> (uniquely mapped read) and the <code>.32</code> (the estimated part) results from an algorithm and can mean multiple things.
</p>
<p>
As mentioned before, we require integer count data for use with packages such as <code>DESeq2</code> and <code>edgeR</code> and there are two methods to convert the expected count to raw count data: + <em>round</em> the value to the nearest integer (widely accepted method and is well within the expected sampling variation), or + discard the fraction part by using for example the <code>floor()</code> function.
</p>
</div>
</div>
<div id="loading-data-1" class="section level3 hasAnchor" number="6.2.3">
<h3><span class="header-section-number">A.2.3</span> Loading Data<a href="a1-data_loading.html#loading-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>From all these columns we want to keep the <em>transcript_id</em> and <em>expected_count</em> columns and ignore the rest (we might be interested in this data later on though). As we need to lead each file separately we can define a function that reads in the data, keeping the columns of interest and returning a dataframe with this data. Note that the first line of each file is used as a header, but check before setting the <code>header</code> argument to <code>TRUE</code>, sometimes the expression data starts at line 1. The file name is then also used to name the column in the dataframe so that we know which column is which sample. This is done by splitting the file name (using <code>strsplit</code>) using the dot (‘<code>.</code>’) keeping the first part (i.e. ‘<code>GSM2970156_HCC1806E224A</code>’) and discarding the second part (‘<code>isoforms.results.txt</code>’). The <code>strsplit</code> function however always returns a list, in this case containing a vector with the 5 splitted elements:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="a1-data_loading.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="do">## String splitting in R</span></span>
<span id="cb44-2"><a href="a1-data_loading.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="do">## (the fixed = TRUE is required as the dot is a special character, see &#39;?strsplit&#39;)</span></span>
<span id="cb44-3"><a href="a1-data_loading.html#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">strsplit</span>(<span class="st">&#39;GSM2970155_HCC1806E224B.isoforms.results.txt.gz&#39;</span>, <span class="st">&#39;.&#39;</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;GSM2970155_HCC1806E224B&quot; &quot;isoforms&quot;               
## [3] &quot;results&quot;                 &quot;txt&quot;                    
## [5] &quot;gz&quot;</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="a1-data_loading.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Keeping the sample identifier</span></span>
<span id="cb46-2"><a href="a1-data_loading.html#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">strsplit</span>(<span class="st">&#39;GSM2970155_HCC1806E224B.isoforms.results.txt.gz&#39;</span>, <span class="st">&#39;.&#39;</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1] &quot;GSM2970155_HCC1806E224B&quot;</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="a1-data_loading.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Function for reading in files</span></span>
<span id="cb48-2"><a href="a1-data_loading.html#cb48-2" aria-hidden="true" tabindex="-1"></a>read_sample <span class="ot">&lt;-</span> <span class="cf">function</span>(file.name) {</span>
<span id="cb48-3"><a href="a1-data_loading.html#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Extract the sample name for naming the column</span></span>
<span id="cb48-4"><a href="a1-data_loading.html#cb48-4" aria-hidden="true" tabindex="-1"></a>  sample.name <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(file.name, <span class="st">&quot;.&quot;</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb48-5"><a href="a1-data_loading.html#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Read the data, setting the &#39;transcript_id&#39; as row.names (column 1)</span></span>
<span id="cb48-6"><a href="a1-data_loading.html#cb48-6" aria-hidden="true" tabindex="-1"></a>  sample <span class="ot">&lt;-</span> <span class="fu">read.table</span>(file.name, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb48-7"><a href="a1-data_loading.html#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Rename the count column</span></span>
<span id="cb48-8"><a href="a1-data_loading.html#cb48-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(sample)[<span class="dv">5</span>] <span class="ot">&lt;-</span> sample.name</span>
<span id="cb48-9"><a href="a1-data_loading.html#cb48-9" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Return a subset containing the &#39;transcript_id&#39; and sample name columns</span></span>
<span id="cb48-10"><a href="a1-data_loading.html#cb48-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sample[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>)])</span>
<span id="cb48-11"><a href="a1-data_loading.html#cb48-11" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Applying the <code>read_sample</code> function to all file names gives us a set of data frames that we can merge together using the <code>merge</code> function. We merge the data based on the transcript id defined with the <code>by = 1</code> argument pointing to the first column. We start by reading in just one file which is the ‘base’ dataframe to which we will merge the other files.</p>
<p>During processing it seemed that this data set is divided into two groups which is also listed on the GEO website for this project:</p>
<ul>
<li>GPL11154 Illumina HiSeq 2000 (Homo sapiens)</li>
<li>GPL13112 Illumina HiSeq 2000 (Mus musculus)</li>
</ul>
<p>where the first 6 files are from human source and the last 6 from the mouse. Therefore, the following code only shows how to read the first 6 samples and merge these into a single dataframe. Repeating this process for the other 6 files would result into another dataframe for those samples.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="a1-data_loading.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&#39;./data/GSE109798_RAW/&#39;</span>)</span>
<span id="cb49-2"><a href="a1-data_loading.html#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="a1-data_loading.html#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Read the FIRST sample</span></span>
<span id="cb49-4"><a href="a1-data_loading.html#cb49-4" aria-hidden="true" tabindex="-1"></a>dataset <span class="ot">&lt;-</span> <span class="fu">read_sample</span>(file.names[<span class="dv">1</span>])</span>
<span id="cb49-5"><a href="a1-data_loading.html#cb49-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-6"><a href="a1-data_loading.html#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Read first sample group (6)</span></span>
<span id="cb49-7"><a href="a1-data_loading.html#cb49-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file.name <span class="cf">in</span> file.names[<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]) {</span>
<span id="cb49-8"><a href="a1-data_loading.html#cb49-8" aria-hidden="true" tabindex="-1"></a>  sample <span class="ot">&lt;-</span> <span class="fu">read_sample</span>(file.name)</span>
<span id="cb49-9"><a href="a1-data_loading.html#cb49-9" aria-hidden="true" tabindex="-1"></a>  dataset <span class="ot">&lt;-</span> <span class="fu">merge</span>(dataset, sample, <span class="at">by =</span> <span class="dv">1</span>)</span>
<span id="cb49-10"><a href="a1-data_loading.html#cb49-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-11"><a href="a1-data_loading.html#cb49-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-12"><a href="a1-data_loading.html#cb49-12" aria-hidden="true" tabindex="-1"></a><span class="fu">pander</span>(<span class="fu">head</span>(dataset))</span></code></pre></div>
<table>
<colgroup>
<col width="20%" />
<col width="26%" />
<col width="26%" />
<col width="27%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">transcript_id</th>
<th align="center">GSM2970149_4T1E274</th>
<th align="center">GSM2970150_4T1E266</th>
<th align="center">GSM2970151_4T1E247D</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">uc007aet.1</td>
<td align="center">1.82</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">uc007aeu.1</td>
<td align="center">0</td>
<td align="center">1</td>
<td align="center">0.24</td>
</tr>
<tr class="odd">
<td align="center">uc007aev.1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">uc007aew.1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0.97</td>
</tr>
<tr class="odd">
<td align="center">uc007aex.2</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">uc007aey.1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
</tbody>
</table>
<p>Table continues below</p>
<table style="width:96%;">
<colgroup>
<col width="31%" />
<col width="31%" />
<col width="31%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">GSM2970152_4T1P2247A</th>
<th align="center">GSM2970153_4T1P2247G</th>
<th align="center">GSM2970154_4T1P2247F</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">0.13</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">0</td>
<td align="center">1</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
</tbody>
</table>
<p>The <code>dataset</code> variable now contains all data for the first 6 samples in this experiment. It is advisable to compare the number of rows in this data set with the number of rows in a single sample. It is not guaranteed that all samples have exactly the same number of genes/transcripts present (i.e., 0-values might have been discarded) which results in a final data set that has as many rows as the <em>smallest</em> sample. See the help of <code>merge</code> if this is the case because the <code>all</code> argument can be used to introduce extra rows for missing data.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter-5.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="a2-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
