<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of Gene Expression</title>
  <meta name="description" content="Course Syllabus for the Theme 7 course">
  <meta name="generator" content="bookdown 0.5.16 and GitBook 2.6.7">

  <meta property="og:title" content="Analysis of Gene Expression" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course Syllabus for the Theme 7 course" />
  <meta name="github-repo" content="mkempenaar/BFVH15CAPSTONE" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Analysis of Gene Expression" />
  
  <meta name="twitter:description" content="Course Syllabus for the Theme 7 course" />
  

<meta name="author" content="Marcel Kempenaar">


<meta name="date" content="2018-02-11">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="chapter-3.html">
<link rel="next" href="eda-part-ii.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Theme 07 - Analysis of Gene Expression</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="chapter-1.html"><a href="chapter-1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter-1.html"><a href="chapter-1.html#theme-7---analysis-of-gene-expression-project"><i class="fa fa-check"></i><b>1.1</b> Theme 7 - Analysis of Gene Expression Project</a></li>
<li class="chapter" data-level="1.2" data-path="chapter-1.html"><a href="chapter-1.html#project-deliverables"><i class="fa fa-check"></i><b>1.2</b> Project Deliverables</a></li>
<li class="chapter" data-level="1.3" data-path="chapter-1.html"><a href="chapter-1.html#project-schedule"><i class="fa fa-check"></i><b>1.3</b> Project Schedule</a></li>
<li class="chapter" data-level="1.4" data-path="chapter-1.html"><a href="chapter-1.html#grading"><i class="fa fa-check"></i><b>1.4</b> Grading</a><ul>
<li class="chapter" data-level="1.4.1" data-path="chapter-1.html"><a href="chapter-1.html#project-proposal"><i class="fa fa-check"></i><b>1.4.1</b> Project Proposal</a></li>
<li class="chapter" data-level="1.4.2" data-path="chapter-1.html"><a href="chapter-1.html#keeping-a-log-a-lab-journal"><i class="fa fa-check"></i><b>1.4.2</b> Keeping a Log (a <em>lab journal</em>)</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chapter-1.html"><a href="chapter-1.html#article"><i class="fa fa-check"></i><b>1.5</b> Article</a><ul>
<li class="chapter" data-level="1.5.1" data-path="chapter-1.html"><a href="chapter-1.html#installing-rstudio-templates"><i class="fa fa-check"></i><b>1.5.1</b> Installing RStudio Templates</a></li>
<li class="chapter" data-level="1.5.2" data-path="chapter-1.html"><a href="chapter-1.html#using-the-template"><i class="fa fa-check"></i><b>1.5.2</b> Using the template</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter-2.html"><a href="chapter-2.html"><i class="fa fa-check"></i><b>2</b> Discovering Data Sets</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter-2.html"><a href="chapter-2.html#finding-public-data"><i class="fa fa-check"></i><b>2.1</b> Finding Public Data Sets of Interest</a></li>
<li class="chapter" data-level="2.2" data-path="chapter-2.html"><a href="chapter-2.html#ncbi-geo"><i class="fa fa-check"></i><b>2.2</b> The NCBI <em>Gene Expression Omnibus</em></a></li>
<li class="chapter" data-level="2.3" data-path="chapter-2.html"><a href="chapter-2.html#project-proposal_ch2"><i class="fa fa-check"></i><b>2.3</b> Creating a Project Proposal</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter-3.html"><a href="chapter-3.html"><i class="fa fa-check"></i><b>3</b> Exploratory Data Analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter-3.html"><a href="chapter-3.html#expl-data"><i class="fa fa-check"></i><b>3.1</b> Exploring the Available Project Data</a></li>
<li class="chapter" data-level="3.2" data-path="chapter-3.html"><a href="chapter-3.html#loading-data"><i class="fa fa-check"></i><b>3.2</b> Loading data into R</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="eda-part-i.html"><a href="eda-part-i.html"><i class="fa fa-check"></i>EDA - Part I</a><ul>
<li class="chapter" data-level="3.3" data-path="eda-part-i.html"><a href="eda-part-i.html#EDA_part1"><i class="fa fa-check"></i><b>3.3</b> Visualizing using <code>summary</code>, <code>boxplot</code>, <code>scatterplot</code> &amp; <code>MA-plot</code></a><ul>
<li class="chapter" data-level="3.3.1" data-path="eda-part-i.html"><a href="eda-part-i.html#eda_statistics"><i class="fa fa-check"></i><b>3.3.1</b> Statistics</a></li>
<li class="chapter" data-level="3.3.2" data-path="eda-part-i.html"><a href="eda-part-i.html#eda_boxplots"><i class="fa fa-check"></i><b>3.3.2</b> Boxplots</a></li>
<li class="chapter" data-level="3.3.3" data-path="eda-part-i.html"><a href="eda-part-i.html#eda_scatter_plot"><i class="fa fa-check"></i><b>3.3.3</b> Scatterplots</a></li>
<li class="chapter" data-level="3.3.4" data-path="eda-part-i.html"><a href="eda-part-i.html#MA_plot"><i class="fa fa-check"></i><b>3.3.4</b> MA-plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="eda-part-ii.html"><a href="eda-part-ii.html"><i class="fa fa-check"></i>EDA - Part II</a><ul>
<li class="chapter" data-level="3.4" data-path="eda-part-ii.html"><a href="eda-part-ii.html#EDA_part2"><i class="fa fa-check"></i><b>3.4</b> Visualizing using <code>heatmap</code>, <code>MDS</code> &amp; <code>PCA</code></a><ul>
<li class="chapter" data-level="3.4.1" data-path="eda-part-ii.html"><a href="eda-part-ii.html#normalization"><i class="fa fa-check"></i><b>3.4.1</b> Normalization</a></li>
<li class="chapter" data-level="3.4.2" data-path="eda-part-ii.html"><a href="eda-part-ii.html#eda_distance"><i class="fa fa-check"></i><b>3.4.2</b> Distance Calculation</a></li>
<li class="chapter" data-level="3.4.3" data-path="eda-part-ii.html"><a href="eda-part-ii.html#eda_heatmap"><i class="fa fa-check"></i><b>3.4.3</b> Sample Distances using a Heatmap</a></li>
<li class="chapter" data-level="3.4.4" data-path="eda-part-ii.html"><a href="eda-part-ii.html#eda_mds"><i class="fa fa-check"></i><b>3.4.4</b> Sample Distances using MDS</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioinf.nl">
Part of the bioinformatics study @ the Hanze University of Applied Sciences</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of Gene Expression</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="eda---part-i" class="section level1 unnumbered">
<h1>EDA - Part I</h1>
<div id="EDA_part1" class="section level2">
<h2><span class="header-section-number">3.3</span> Visualizing using <code>summary</code>, <code>boxplot</code>, <code>scatterplot</code> &amp; <code>MA-plot</code></h2>
<p>This segment describes some of the basic steps and visualizations that can be performed during Exploratory Data Analysis. In this part we work with the unnormalized count data. In part this data is normalized before analyzing it further. As mentioned above, the focus of EDA is to get an overview of the data set and while this often requires visualizing the data, these figures do not need to be very pretty. Simple figures are perfectly fine in this stage. Try to create these figures for your own data (and keep them in your log).</p>
<p>Also note that these steps are just a selection. Furthermore, make sure that for every visualization you make, add proper axis-labels containing the measurement units (important!).</p>
<div id="eda_statistics" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Statistics</h3>
<p>Even the most basic statistics can give some insight into the data such as performing a 6-number-statistic on the data <strong>columns</strong> using the <code>summary()</code> function.</p>
</div>
<div id="eda_boxplots" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Boxplots</h3>
<p>A visual representation of these values can be shown in a boxplot. Boxplots are very easy to create from an R data frame object by just passing in the data columns. The following boxplot shows the data for the complete experiment with a separate boxplot for each sample.</p>
<div class="figure">
<img src="chapters/images/RNAboxplot.png" alt="Boxplot comparing basic statistics for all genes across multiple samples" />
<p class="caption"><em>Boxplot comparing basic statistics for all genes across multiple samples</em></p>
</div>
</div>
<div id="eda_scatter_plot" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Scatterplots</h3>
<p>Scatterplots are good for spotting correlations between two sets of numerical data. Data that is similar will show up as a diagonal line in a scatter plot and the more they differ, the bigger the spread of points relative to the diagonal. Downside is that a scatterplot can only contain data from two columns and it <em>might</em> not always be useful to plot all samples from the condition group to all samples from the control group. Other possibilities include each sample against the <strong>average</strong> of its group (should be fairly diagonal), the averages of both groups in one scatterplot (this probably shows a fair amount of variation depending on the experiment).</p>
<p>A scatterplot can be made simply using the <code>plot()</code> functions with the <code>type</code> argument set to <code>p</code> (points). To show the correlation more clearly you can add a <em>linear regression</em> line using <code>abline(lm(y ~ x, col='red', lwd=2)</code>. If the regression line points more downwards from the diagonal, the values in sample <code>x</code> are higher and vice versa.</p>
<div class="figure">
<img src="chapters/images/RNAscatterplot.png" alt="A simple scatterplot showing the correlation between two samples or experiments (see the text above). This image shows the values log2 transformed which in most cases gives a clearer picture." />
<p class="caption"><em>A simple scatterplot showing the correlation between two samples or experiments (see the text above). This image shows the values log2 transformed which in most cases gives a clearer picture.</em></p>
</div>
</div>
<div id="MA_plot" class="section level3">
<h3><span class="header-section-number">3.3.4</span> MA-plots</h3>
<p>An <strong>MA-plot</strong> is a scatterplot comparing the average expression value per gene (<strong>A</strong>) against the log-fold-change (<strong>M</strong>; a log2 value indicating expression change across samples which is taken as the <code>log2(sample1 / sample2))</code>). The points in this plot <em>should</em> be centered around the horizontal 0-value with an even spread both above and below the line. If there are (far) more points above or below the 0 on the y-axis, there is a <em>bias</em> in the data.</p>
<p>An MA-plot can be very helpful (it has the same constraints mentioned with the scatterplot) but might not be easy/ possible to create as this is mostly done on <em>normalized count data</em> other than FPKM or RPKM. If you have raw or normalized count data in your data set you can search various <a href="https://en.wikipedia.org/wiki/MA_plot">online resources</a> on how to create this plot.</p>
<div class="figure">
<img src="chapters/images/RNAmaplot.png" alt="An MA-plot showing the average normalized count values (A, x-axis) vs. the log2 fold-change (M, y-axis)." />
<p class="caption"><em>An MA-plot showing the average normalized count values (<strong>A</strong>, x-axis) vs. the log2 fold-change (<strong>M</strong>, y-axis).</em></p>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter-3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="eda-part-ii.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"download": ["_main.pdf", "_main.epub", "_main.mobi"],
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
