<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> Mapping with STAR | 2.1.2 Genomics &amp; Transcriptomics</title>
  <meta name="description" content="Course Syllabus for course 2.1.2" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content=" Mapping with STAR | 2.1.2 Genomics &amp; Transcriptomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course Syllabus for course 2.1.2" />
  <meta name="github-repo" content="mkempenaar/gene_expression_analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" Mapping with STAR | 2.1.2 Genomics &amp; Transcriptomics" />
  
  <meta name="twitter:description" content="Course Syllabus for course 2.1.2" />
  

<meta name="author" content="Marcel Kempenaar" />


<meta name="date" content="2023-10-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="from-reads-to-counts.html"/>
<link rel="next" href="chapter-3.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">2.1.2 Genomics & Transcriptomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genomics"><i class="fa fa-check"></i><b>1.1</b> 2.1.2 Genomics</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#project-deliverables"><i class="fa fa-check"></i><b>1.1.1</b> Project Deliverables</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#transcriptomics"><i class="fa fa-check"></i><b>1.2</b> 2.1.2 Transcriptomics</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#project-deliverables-1"><i class="fa fa-check"></i><b>1.2.1</b> Project Deliverables</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#lab-journal"><i class="fa fa-check"></i><b>1.3</b> <em>Lab Journal</em></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.4</b> Schedule</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#genomics-1"><i class="fa fa-check"></i><b>1.4.1</b> Genomics</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#transcriptomics-1"><i class="fa fa-check"></i><b>1.4.2</b> Transcriptomics</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#literature"><i class="fa fa-check"></i><b>1.5</b> Literature</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="from-reads-to-counts.html"><a href="from-reads-to-counts.html"><i class="fa fa-check"></i><b>2</b> From Reads to Counts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="from-reads-to-counts.html"><a href="from-reads-to-counts.html#downloading-and-extracting-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Downloading and extracting FastQ files</a></li>
<li class="chapter" data-level="2.2" data-path="from-reads-to-counts.html"><a href="from-reads-to-counts.html#optional-annotation"><i class="fa fa-check"></i><b>2.2</b> (Optional) Annotation</a></li>
<li class="chapter" data-level="2.3" data-path="from-reads-to-counts.html"><a href="from-reads-to-counts.html#quality-control"><i class="fa fa-check"></i><b>2.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="from-reads-to-counts.html"><a href="from-reads-to-counts.html#fastqc"><i class="fa fa-check"></i><b>2.3.1</b> FastQC</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="from-reads-to-counts.html"><a href="from-reads-to-counts.html#trimming"><i class="fa fa-check"></i><b>2.4</b> Trimming</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mapping-with-star.html"><a href="mapping-with-star.html"><i class="fa fa-check"></i><b>3</b> Mapping with STAR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="mapping-with-star.html"><a href="mapping-with-star.html#building-star-index"><i class="fa fa-check"></i><b>3.1</b> Building STAR index</a></li>
<li class="chapter" data-level="3.2" data-path="mapping-with-star.html"><a href="mapping-with-star.html#read-mapping-and-transcript-counting"><i class="fa fa-check"></i><b>3.2</b> Read mapping and transcript counting</a></li>
<li class="chapter" data-level="3.3" data-path="mapping-with-star.html"><a href="mapping-with-star.html#combining-star-count-files"><i class="fa fa-check"></i><b>3.3</b> Combining STAR Count Files</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter-3.html"><a href="chapter-3.html"><i class="fa fa-check"></i><b>4</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="chapter-3.html"><a href="chapter-3.html#loading-data"><i class="fa fa-check"></i><b>4.1</b> Loading data into R</a></li>
<li class="chapter" data-level="4.2" data-path="chapter-3.html"><a href="chapter-3.html#ch3_example_data"><i class="fa fa-check"></i><b>4.2</b> Example Data</a></li>
<li class="chapter" data-level="4.3" data-path="chapter-3.html"><a href="chapter-3.html#EDA_part1"><i class="fa fa-check"></i><b>4.3</b> Visualizing using <code>boxplot</code> and <code>density plot</code></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="chapter-3.html"><a href="chapter-3.html#eda_statistics"><i class="fa fa-check"></i><b>4.3.1</b> Statistics</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter-3.html"><a href="chapter-3.html#eda_boxplots"><i class="fa fa-check"></i><b>4.3.2</b> Boxplots</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter-3.html"><a href="chapter-3.html#eda_density_plot"><i class="fa fa-check"></i><b>4.3.3</b> Density Plots</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter-3.html"><a href="chapter-3.html#EDA_part2"><i class="fa fa-check"></i><b>4.4</b> Visualizing using <code>heatmap</code> and <code>MDS</code></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="chapter-3.html"><a href="chapter-3.html#normalization"><i class="fa fa-check"></i><b>4.4.1</b> Normalization</a></li>
<li class="chapter" data-level="4.4.2" data-path="chapter-3.html"><a href="chapter-3.html#eda_distance"><i class="fa fa-check"></i><b>4.4.2</b> Distance Calculation</a></li>
<li class="chapter" data-level="4.4.3" data-path="chapter-3.html"><a href="chapter-3.html#eda_heatmap"><i class="fa fa-check"></i><b>4.4.3</b> Sample Distances using a Heatmap</a></li>
<li class="chapter" data-level="4.4.4" data-path="chapter-3.html"><a href="chapter-3.html#eda_mds"><i class="fa fa-check"></i><b>4.4.4</b> Multi-Dimensional Scaling</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="chapter-3.html"><a href="chapter-3.html#cleaning-data"><i class="fa fa-check"></i><b>4.5</b> Cleaning Data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter-4.html"><a href="chapter-4.html"><i class="fa fa-check"></i><b>5</b> Discovering Differentialy Expressed Genes (DEGs)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="chapter-4.html"><a href="chapter-4.html#deg-preprocessing"><i class="fa fa-check"></i><b>5.1</b> Pre-processing</a></li>
<li class="chapter" data-level="5.2" data-path="chapter-4.html"><a href="chapter-4.html#deg-fc"><i class="fa fa-check"></i><b>5.2</b> The <em>Fold Change</em> Value</a></li>
<li class="chapter" data-level="5.3" data-path="chapter-4.html"><a href="chapter-4.html#deg-bioconductor"><i class="fa fa-check"></i><b>5.3</b> Using Bioconductor Packages</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="chapter-4.html"><a href="chapter-4.html#deg-design"><i class="fa fa-check"></i><b>5.3.1</b> The <code>Design</code> (matrix)</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter-4.html"><a href="chapter-4.html#deg-deseq2"><i class="fa fa-check"></i><b>5.3.2</b> <code>DESeq2</code></a></li>
<li class="chapter" data-level="5.3.3" data-path="chapter-4.html"><a href="chapter-4.html#deg-edger"><i class="fa fa-check"></i><b>5.3.3</b> <code>edgeR</code></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter-5.html"><a href="chapter-5.html"><i class="fa fa-check"></i><b>6</b> Data Analysis and Visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="chapter-5.html"><a href="chapter-5.html#volcano"><i class="fa fa-check"></i><b>6.1</b> Volcano Plot</a></li>
<li class="chapter" data-level="6.2" data-path="chapter-5.html"><a href="chapter-5.html#venn-diagram"><i class="fa fa-check"></i><b>6.2</b> Venn Diagram</a></li>
<li class="chapter" data-level="6.3" data-path="chapter-5.html"><a href="chapter-5.html#clustering"><i class="fa fa-check"></i><b>6.3</b> Clustering</a></li>
<li class="chapter" data-level="6.4" data-path="chapter-5.html"><a href="chapter-5.html#pathway-analysis"><i class="fa fa-check"></i><b>6.4</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="chapter-5.html"><a href="chapter-5.html#gene-functional-classification-using-david-100-degs"><i class="fa fa-check"></i><b>6.4.1</b> Gene Functional Classification using DAVID (&gt;100 DEGs)</a></li>
<li class="chapter" data-level="6.4.2" data-path="chapter-5.html"><a href="chapter-5.html#kegg-pathway-visualization-10-degs"><i class="fa fa-check"></i><b>6.4.2</b> KEGG-pathway Visualization (&gt;10 DEGs)</a></li>
<li class="chapter" data-level="6.4.3" data-path="chapter-5.html"><a href="chapter-5.html#spia"><i class="fa fa-check"></i><b>6.4.3</b> Signaling Pathway Impact Analysis (&gt;20 DEGs)</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="a1-data_loading.html"><a href="a1-data_loading.html"><i class="fa fa-check"></i><b>A</b> Appendix A: Loading Expression Data in R</a>
<ul>
<li class="chapter" data-level="A.1" data-path="a1-data_loading.html"><a href="a1-data_loading.html#expl-data"><i class="fa fa-check"></i><b>A.1</b> Exploring the Available Project Data</a></li>
<li class="chapter" data-level="A.2" data-path="a1-data_loading.html"><a href="a1-data_loading.html#a1-batch_data_loading"><i class="fa fa-check"></i><b>A.2</b> Batch Loading Expression Data</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="a1-data_loading.html"><a href="a1-data_loading.html#decompressing"><i class="fa fa-check"></i><b>A.2.1</b> Decompressing</a></li>
<li class="chapter" data-level="A.2.2" data-path="a1-data_loading.html"><a href="a1-data_loading.html#determining-data-format"><i class="fa fa-check"></i><b>A.2.2</b> Determining Data Format</a></li>
<li class="chapter" data-level="A.2.3" data-path="a1-data_loading.html"><a href="a1-data_loading.html#loading-data-1"><i class="fa fa-check"></i><b>A.2.3</b> Loading Data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="a2-annotation.html"><a href="a2-annotation.html"><i class="fa fa-check"></i><b>B</b> Annotating an RNA-Seq Experiment</a>
<ul>
<li class="chapter" data-level="B.1" data-path="a2-annotation.html"><a href="a2-annotation.html#download-geo"><i class="fa fa-check"></i><b>B.1</b> Downloading annotation data from GEO</a></li>
<li class="chapter" data-level="B.2" data-path="a2-annotation.html"><a href="a2-annotation.html#manually-annotate"><i class="fa fa-check"></i><b>B.2</b> Manual Data Annotation</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="a2-annotation.html"><a href="a2-annotation.html#annotationDBI"><i class="fa fa-check"></i><b>B.2.1</b> Using <code>AnnotationDBI</code></a></li>
<li class="chapter" data-level="B.2.2" data-path="a2-annotation.html"><a href="a2-annotation.html#using-biomart"><i class="fa fa-check"></i><b>B.2.2</b> Using <code>biomaRt</code></a></li>
</ul></li>
<li class="chapter" data-level="B.3" data-path="a2-annotation.html"><a href="a2-annotation.html#references"><i class="fa fa-check"></i><b>B.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="a3-annotation.html"><a href="a3-annotation.html"><i class="fa fa-check"></i><b>C</b> Discovering Data Sets</a>
<ul>
<li class="chapter" data-level="C.1" data-path="a3-annotation.html"><a href="a3-annotation.html#finding-public-data"><i class="fa fa-check"></i><b>C.1</b> Finding Public Data Sets of Interest</a></li>
<li class="chapter" data-level="C.2" data-path="a3-annotation.html"><a href="a3-annotation.html#ncbi-geo"><i class="fa fa-check"></i><b>C.2</b> The NCBI <em>Gene Expression Omnibus</em></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">2.1.2 Genomics &amp; Transcriptomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mapping-with-star" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number"></span> Mapping with STAR<a href="mapping-with-star.html#mapping-with-star" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="building-star-index" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Building STAR index<a href="mapping-with-star.html#building-star-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Other than with <code>bwa</code>, STAR requires an extra annotation file (GFF or GTF format) describing features on the genome. This data set requires the FASTA of the Arabidopsis thaliana genome and its accompanying annotation file. The <code>genomeDir</code> specifies the output location for the index.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="mapping-with-star.html#cb9-1" tabindex="-1"></a><span class="bu">cd</span> /local-fs/staff/marcel/GSE149995/reference</span>
<span id="cb9-2"><a href="mapping-with-star.html#cb9-2" tabindex="-1"></a><span class="fu">mkdir</span> star<span class="kw">;</span> <span class="bu">cd</span> star</span>
<span id="cb9-3"><a href="mapping-with-star.html#cb9-3" tabindex="-1"></a><span class="ex">STAR</span> <span class="dt">\</span></span>
<span id="cb9-4"><a href="mapping-with-star.html#cb9-4" tabindex="-1"></a>  <span class="at">--runThreadN</span> 12 <span class="dt">\</span></span>
<span id="cb9-5"><a href="mapping-with-star.html#cb9-5" tabindex="-1"></a>  <span class="at">--runMode</span> genomeGenerate <span class="dt">\</span></span>
<span id="cb9-6"><a href="mapping-with-star.html#cb9-6" tabindex="-1"></a>  <span class="at">--genomeDir</span> /local-fs/staff/marcel/GSE149995/reference/star <span class="dt">\</span></span>
<span id="cb9-7"><a href="mapping-with-star.html#cb9-7" tabindex="-1"></a>  <span class="at">--genomeFastaFiles</span> /local-fs/staff/marcel/GSE149995/reference/TAIR10_chr_all.fas <span class="dt">\</span></span>
<span id="cb9-8"><a href="mapping-with-star.html#cb9-8" tabindex="-1"></a>  <span class="at">--sjdbGTFfile</span> /local-fs/staff/marcel/GSE149995/reference/TAIR10_GTF_genes.gtf <span class="dt">\</span></span>
<span id="cb9-9"><a href="mapping-with-star.html#cb9-9" tabindex="-1"></a>  <span class="at">--genomeSAindexNbases</span> 12 <span class="dt">\</span></span>
<span id="cb9-10"><a href="mapping-with-star.html#cb9-10" tabindex="-1"></a>  <span class="at">--sjdbOverhang</span> 49</span></code></pre></div>
</div>
<div id="read-mapping-and-transcript-counting" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Read mapping and transcript counting<a href="mapping-with-star.html#read-mapping-and-transcript-counting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The following command performs mapping against the reference genome, saves this as a sorted BAM file and directly performs a read count for all transcripts (NOTE: requires the <code>--sjdbGTFfile</code> parameter during index building). For each input file, there is a <code>ReadsPerGenes.out.tab</code> file generated containing 4 columns:
* column 1: gene ID
* column 2: counts for unstranded RNA-seq
* column 3: counts for the 1st read strand aligned with RNA (htseq-count option -s yes)
* column 4: counts for the 2nd read strand aligned with RNA (htseq-count option -s reverse)</p>
<p>See an interesting discussion on <a href="https://www.biostars.org/p/218995/">BioStars</a> regarding which column to use.</p>
<p>When planning to use a different tool for calculating gene counts, you can remove the <code>quantMode</code> argument which will only generate the sorted BAM files. Please read the manual of the tool of choice to see if an indexed BAM file as the BAM file does not have an index.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="mapping-with-star.html#cb10-1" tabindex="-1"></a><span class="fu">find</span> /local-fs/staff/marcel/GSE149995/fastq-trimmed/ <span class="at">-name</span> <span class="st">&quot;*.fastq.gz&quot;</span> <span class="at">-exec</span> basename {} <span class="dt">\;</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb10-2"><a href="mapping-with-star.html#cb10-2" tabindex="-1"></a>  <span class="fu">sed</span> <span class="st">&#39;s/\.fastq.gz$//&#39;</span> <span class="kw">|</span> <span class="ex">parallel</span> <span class="st">&#39;STAR --runThreadN 6 &#39;</span> <span class="dt">\</span></span>
<span id="cb10-3"><a href="mapping-with-star.html#cb10-3" tabindex="-1"></a>    <span class="st">&#39;--genomeDir /local-fs/staff/marcel/GSE149995/reference/star &#39;</span> <span class="dt">\</span></span>
<span id="cb10-4"><a href="mapping-with-star.html#cb10-4" tabindex="-1"></a>    <span class="st">&#39;--readFilesIn /local-fs/staff/marcel/GSE149995/fastq-trimmed/{}.fastq &#39;</span> <span class="dt">\</span></span>
<span id="cb10-5"><a href="mapping-with-star.html#cb10-5" tabindex="-1"></a>    <span class="st">&#39;--outSAMtype BAM SortedByCoordinate &#39;</span> <span class="dt">\</span></span>
<span id="cb10-6"><a href="mapping-with-star.html#cb10-6" tabindex="-1"></a>    <span class="st">&#39;--quantMode GeneCounts &#39;</span> <span class="dt">\</span></span>
<span id="cb10-7"><a href="mapping-with-star.html#cb10-7" tabindex="-1"></a>    <span class="st">&#39;--outFileNamePrefix /local-fs/staff/marcel/GSE149995/mapping/star/{}_star_&#39;</span></span></code></pre></div>
<p>Generating MultiQC report for STAR mapping phase</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="mapping-with-star.html#cb11-1" tabindex="-1"></a><span class="ex">~/miniconda3/bin/multiqc</span> <span class="at">--filename</span> <span class="dt">\</span></span>
<span id="cb11-2"><a href="mapping-with-star.html#cb11-2" tabindex="-1"></a>  ~/Development/2.1.2-Transcriptomics/STAR-mapping.html <span class="dt">\</span></span>
<span id="cb11-3"><a href="mapping-with-star.html#cb11-3" tabindex="-1"></a>  /local-fs/staff/marcel/GSE149995/mapping/star/</span></code></pre></div>
</div>
<div id="combining-star-count-files" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Combining STAR Count Files<a href="mapping-with-star.html#combining-star-count-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>STAR generates a <code>[sample]_star_ReadsPerGene.out.tab</code> file for each sample, containing the gene IDs and the count numbers. We can merge these files into a single data frame for further processing (see also <a href="https://mkempenaar.github.io/gene_expression_analysis/a1-data_loading.html#a1-batch_data_loading">Appendix A1 in the manual</a>). This example takes the 2nd column for the count values. Make sure that you know which column to use based on the description above. The reason that weâ€™re merging the data is because it might not always be guaranteed that you get the exact same transcripts in each file.</p>
<p>Note: if you use the <code>featurecounts</code> tool, you will already get a single file containing the count values for all samples.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="mapping-with-star.html#cb12-1" tabindex="-1"></a>file.names <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&#39;/local-fs/staff/marcel/GSE149995/mapping/star/&#39;</span>,</span>
<span id="cb12-2"><a href="mapping-with-star.html#cb12-2" tabindex="-1"></a>                         <span class="at">pattern =</span> <span class="st">&#39;*_star_ReadsPerGene.out.tab&#39;</span>)</span>
<span id="cb12-3"><a href="mapping-with-star.html#cb12-3" tabindex="-1"></a></span>
<span id="cb12-4"><a href="mapping-with-star.html#cb12-4" tabindex="-1"></a><span class="do">## Function for reading in files</span></span>
<span id="cb12-5"><a href="mapping-with-star.html#cb12-5" tabindex="-1"></a>read_sample <span class="ot">&lt;-</span> <span class="cf">function</span>(file.name) {</span>
<span id="cb12-6"><a href="mapping-with-star.html#cb12-6" tabindex="-1"></a>  <span class="do">## Extract the sample name for naming the column (retaining the &#39;SRR....&#39; part)</span></span>
<span id="cb12-7"><a href="mapping-with-star.html#cb12-7" tabindex="-1"></a>  sample.name <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(file.name, <span class="st">&quot;.&quot;</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb12-8"><a href="mapping-with-star.html#cb12-8" tabindex="-1"></a>  sample <span class="ot">&lt;-</span> <span class="fu">read.table</span>(file.name, <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, </span>
<span id="cb12-9"><a href="mapping-with-star.html#cb12-9" tabindex="-1"></a>                       <span class="at">row.names =</span> <span class="cn">NULL</span>, <span class="at">skip =</span> <span class="dv">4</span>)</span>
<span id="cb12-10"><a href="mapping-with-star.html#cb12-10" tabindex="-1"></a>  <span class="do">## Rename the count column</span></span>
<span id="cb12-11"><a href="mapping-with-star.html#cb12-11" tabindex="-1"></a>  <span class="fu">names</span>(sample)[<span class="dv">2</span>] <span class="ot">&lt;-</span> sample.name</span>
<span id="cb12-12"><a href="mapping-with-star.html#cb12-12" tabindex="-1"></a>  <span class="do">## Return a subset containing the transcript ID and sample name columns</span></span>
<span id="cb12-13"><a href="mapping-with-star.html#cb12-13" tabindex="-1"></a>  <span class="fu">return</span>(sample[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)])</span>
<span id="cb12-14"><a href="mapping-with-star.html#cb12-14" tabindex="-1"></a>}</span>
<span id="cb12-15"><a href="mapping-with-star.html#cb12-15" tabindex="-1"></a></span>
<span id="cb12-16"><a href="mapping-with-star.html#cb12-16" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&#39;/local-fs/staff/marcel/GSE149995/mapping/star/&#39;</span>)</span>
<span id="cb12-17"><a href="mapping-with-star.html#cb12-17" tabindex="-1"></a></span>
<span id="cb12-18"><a href="mapping-with-star.html#cb12-18" tabindex="-1"></a><span class="do">## Read the FIRST sample</span></span>
<span id="cb12-19"><a href="mapping-with-star.html#cb12-19" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read_sample</span>(file.names[<span class="dv">1</span>])</span>
<span id="cb12-20"><a href="mapping-with-star.html#cb12-20" tabindex="-1"></a></span>
<span id="cb12-21"><a href="mapping-with-star.html#cb12-21" tabindex="-1"></a><span class="do">## Read the remaining files and merge the contents</span></span>
<span id="cb12-22"><a href="mapping-with-star.html#cb12-22" tabindex="-1"></a><span class="cf">for</span> (file.name <span class="cf">in</span> file.names[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(file.names)]) {</span>
<span id="cb12-23"><a href="mapping-with-star.html#cb12-23" tabindex="-1"></a>  sample <span class="ot">&lt;-</span> <span class="fu">read_sample</span>(file.name)</span>
<span id="cb12-24"><a href="mapping-with-star.html#cb12-24" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">merge</span>(counts, sample, <span class="at">by =</span> <span class="dv">1</span>)</span>
<span id="cb12-25"><a href="mapping-with-star.html#cb12-25" tabindex="-1"></a>}</span>
<span id="cb12-26"><a href="mapping-with-star.html#cb12-26" tabindex="-1"></a></span>
<span id="cb12-27"><a href="mapping-with-star.html#cb12-27" tabindex="-1"></a><span class="co"># Set the row names to the transcript IDs</span></span>
<span id="cb12-28"><a href="mapping-with-star.html#cb12-28" tabindex="-1"></a><span class="fu">rownames</span>(counts) <span class="ot">&lt;-</span> counts<span class="sc">$</span>V1</span>
<span id="cb12-29"><a href="mapping-with-star.html#cb12-29" tabindex="-1"></a>counts <span class="ot">&lt;-</span> counts[<span class="sc">-</span><span class="dv">1</span>]</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="from-reads-to-counts.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter-3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
