# Introduction {#chapter-1}

## Course changes due to COVID restrictions

Even though restrictions are lifted, the manual will not be restored to the pre-COVID period. The following sections have been **removed** from the manual and are now regarded as optional for 2022. The following listing contains links to the sections on a backup-site where the contents are still available. The complete previous version of this manual is available at [https://bioinf.nl/~marcelk/theme07/](https://bioinf.nl/~marcelk/theme07/). Please **do** read the parts (once you are working on the respective chapters) that have been removed as they might contain information that is helpful for the understanding of the remaining parts.

* [3.4.4 Scatterplots](https://bioinf.nl/~marcelk/theme07/chapter-3.html#eda_scatter_plot) - this plot can only compare between single samples and is better to do *after* we have our DEGs comparing the average expression of replicates between groups.
* [3.4.5 MA-plot](https://bioinf.nl/~marcelk/theme07/chapter-3.html#MA_plot) - same as with the scatterplots; we will most likely create this plot using the identified DEGs as part of chapter 5.
* [3.5.5 Principal Component Analysis](https://bioinf.nl/~marcelk/theme07/chapter-3.html#eda_pca) - comparable with MDS and therefore superfluous
* [4.3 Using Manual Methods (t-test & ANOVA)](https://bioinf.nl/~marcelk/theme07/chapter-4.html#deg-manual) - we will now only focus on finding our DEGs using the packages described in chapter 4, the manual analysis - while a valuable exercise - is often not used in the end conclusions
* [4.4 Multiple Testing Correction](https://bioinf.nl/~marcelk/theme07/chapter-4.html#deg-mtc) - only needed when doing manual testing, therefore removed.

## Theme 7 - Analysis of Gene Expression Project

The use of RNA-Sequencing techniques for measuring gene expression is relatively new and replaces *microarrays*, though in some cases microarrays are still used. Gene expression data gives valuable insights into the workings of cells in certain conditions. Especially when comparing for instance healthy and diseased samples it can become clear which genes are causal or under influence of a specific condition. Finding the genes of interest (genes showing differing expression accross conditions, called the Differentially Expressed Genes (DEGs)) is the goal of this project.

While there is no golden standard for analyzing RNA-sequencing data sets as there are many tools (all manufacturers of sequencing equiptment also deliver software packages) we will use proven R libraries for processing, visualizing and analyzing publically available data sets. While in some cases you are allowed to use the actual *raw* data (sequence reads, in FastQ format) that is available, it is highly recommended to use the pre-processed data which often is a table with a *count* value for each gene. This count is the number of reads that was mapped to that gene which corresponds to the relative number of transcripts (mRNA sequences) of that gene present in the cell at the time of sampling.

## Project Deliverables

The end products of this course consist of two deliverables; a PDF output file from an RMarkdown 'lab journal' where you have logged all steps performed to get to the end result and a poster to be presented - if selected - during the Life Science and Technology Poster session at the end of this quarter.

## Project Schedule

The aim is to keep to the below schedule during this course. Use the first two weeks to see if you need to focus more on one of the points below and discuss changes to the planning with your teacher.

* Find a public experiment of interest [week one]
    + Using online resources (sections [2.1](#finding-public-data), [2.2](#ncbi-geo))
* Data Gathering and Literature Research [week three]
    + Make a final project choice
    + Retrieve the accompanying publication
    + Write a short project proposal (section [2.3](#project-proposal_ch2))
    + Starting with Exploratory Data Analysis (chapter [3](#chapter-3))
* Data Analysis [weeks four and five]
    + Finalizing Exploratory Data Analysis
    + Discovering Differentially Expressed Genes (chapter [4](#chapter-4))
    + Data Annotation (**optional**; appendix [a2](#a2-annotation))
    + Techniques used: R with bioconductor, the edgeR and/ or DESeq2 packages
* Result Analysis [week six]
    + Analyzing and Visualizing your results (chapter [5](#chapter-5))
    + Techniques used: clustering, pathway analysis, gene-enrichment analysis
* Finalizing analysis and designing/ presenting a poster [weeks 7 and 8]

## Grading {#grading}

The final grade consists of a weighted average of the *lab journal* containing all performed steps, their code and outputs (**70%**), the created poster (**20%**) and finally your *work attitude* (**10%**). A grade higher then a **5.5** will give a total of 6 EC. The rest of this chapter explains the expected contents for the project proposal and the two graded products (each of these elements is explained in greater detail in other chapters).

## Project Proposal

As you are free to choose from thousands of public data sets on a large variety of biological subjects, it is important to demonstrate a good understanding of the experiment, biology and the available data. This is demonstrated by writing a short project proposal where you briefly explain the subject of the chosen research article, the experimental setup (how many samples were used, etc.) and what types of data are available for you to use.

## *Lab Journal*

As you know from previous projects, it is essential to keep a proper journal detailing every step you have done during the analysis. This journal is to be kept in an R markdown file, showing which steps have been taken in the analysis of the data set. This markdown should be *knitted* into a single PDF-file once the project is completed thus containing text detailling the steps and any decisions you've made, R-code (always visible!) and their resulting output/ images. 

```{block, type='rmdtip'}
**Notes**:
As a general advice; do not wait with knitting this whole document until the project is done as knitting is very prone to errors and trying to fix these in a large document is not easy. Give each code chunk the proper attributes, including a name at the minimum. This helps spot errors during knitting as that process mentiones which chunk has been processed. Note that chunk-names *must* be unique. And try to make proper use of chapters and sections and include an (optional) table of contents.
```