# Introduction {#chapter-1}

## Capstone Project

This project aims to teach the whole process of analysing a dataset containing gene expression data measured with the RNA-sequencing technique. The tools we will use to perform the analysis will be the statistical programming language `R` with its many libraries and the software package [RStudio](https://www.rstudio.com) to interact with R. Rstudio is a so called Integrated Development Environment (IDE) which makes programming in R much easier.

The use of RNA-Sequencing techniques for measuring gene expression is relatively new and replaces *microarrays*, though in some cases microarrays are still used. Gene expression data gives valuable insights into the workings of cells in certain conditions. Especially when comparing for instance healthy and diseased samples it can become clear which genes are causal or under influence of a specific condition. Finding the genes of interest (genes showing differing expression accross conditions, called the Differentially Expressed Genes (DEGs)) is the goal of this project.

While there is no golden standard for analyzing RNA-sequencing datasets as there are many tools (all manufacturers of sequencing equiptment also deliver software packages) we will use proven R libraries for processing, visualizing and analyzing publically available datasets. While in some cases you are allowed to use the actual *raw* data that is available, it is highly recommended to use the pre-processed data which often is a table with a *count* value for each gene. This count is the number of reads that was mapped to that gene which corresponds to the relative number of transcripts (mRNA sequences) of that gene present in the cell at the time of sampling.

## Finding Data {#finding-data}

Our first task is to search through the large collections of potential datasets. These can be found in multiple online databases such as NCBI's [GEO](https://www.ncbi.nlm.nih.gov/geo/) and [SRA](https://www.ncbi.nlm.nih.gov/sra), the ENCODE [RNA Dashboard](http://genome.crg.es/~jlagarde/encode_RNA_dashboard/), The Cancer Genome Atlas ([TCGA](https://tcga-data.nci.nih.gov/docs/publications/tcga/?)) and probably many more. 

## Analyzing an RNA-Seq Data Set

Once a suitable data set has been found we will fully analyze the data using the programming language [R](https://www.r-project.org) combined with libraries from the [Bioconductor](https://www.bioconductor.org) collection.

## Project Deliverables

The end product of this course consists of two deliverables; a PDF output file from an RMarkdown 'lab journal' where you have logged all steps performed to get to the end result and a final report in the form of a short article. This chapter describes the requirements and contents of each of these products and shows how to use an RMarkdown `template` that can be a useful guide for managing your content.

## Grading {#grading}

The final grade consists of a weighted average of the work done for weeks 1 & 2 (*Discovering Data Sets*) (**10%**), the *lab journal* (**50%**), the *article* (**30%**) report and finally your *work attitude* (**10%**). A grade higher then a **5.5** will give a total of 8EC. The rest of this chapter explains the expected contents for the last three graded items.

## Keeping a Log (a *lab journal*)

As you know from working in the laboratory it is essential to keep a proper lab journal detailing every step you have done during the experiment. The log is to be kept in an R markdown file, showing which steps have been taken in the analysis of the data set. This markdown should be *knitted* into a single PDF-file once the project is completed.

## Article {#article}

The final report is written in article form which is a bit different from a usual report mainly in its size. The article-report has a *maximum* number of pages of **6** including all images and references (no appendices!). 

### Installing RStudio Templates

The templates are available in an R package and contains both `RMarkdown` and \LaTeX (another layer on top of the actual [\TeX markup language](https://nl.wikipedia.org/wiki/TeX), yes, another language...) files. RStudio can use templates for a number of documents, including article-templates. These templates can be installed from a package called `rticles` by running the `install.packages` function:

```{r install_rticles, eval=FALSE}
install.packages("rticles", type = "source")
```

### Using the template

Now that you have the templates package (`rticles`) installed you can download and use the template project (ZIP-file) available from [http://bioinf.nl/~marcelk/capstone/report-template.zip](http://bioinf.nl/~marcelk/capstone/report-template.zip). Download this file to your project folder, extract its contents and open the `report-template.Rmd` file contained within the folder in RStudio. Verify that everything is setup correctly by hitting the `Knit` button at the top, this should create a PDF version of the report. Note that - somehow - the resulting PDF file is named `RJwrapper.pdf` instead of the expected `report-template.pdf`. If everything checks out ok you can rename the file to your liking and start editing.

This template is based on the `R Journal Submission` template that you can also find in RStudio in the `New file -> R Markdown -> From Template` menu. Articles published in the `R-journal` are based on this template which you can browse for inspiration at the [R Journal Website]((https://journal.r-project.org/archive/2015-2/). 

The available template shows an *example* of segments/ chapters and briefly describes what each section could or should contain. If you want to write your report in the Dutch language, you can create a new file from the template and change the segment names and content to Dutch. There will however still be some headings added by the template in English which is fine with me, but if you want to keep everything Dutch you can either edit the `RJournal.sty` file manually (*not recommended*) or start a new file by yourself and add some nice headings and page options. 

The top part of the template (between the dashes `---`) contains some settings that you need to change such as `title`, `author` and `abstract`. Compare for instance a newly created article from this template with the one offered at the `bioinf.nl` website.