<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title> Introduction | 2.1.2 Genomics &amp; Transcriptomics</title>
<meta name="author" content="Marcel Kempenaar">
<meta name="description" content="Course Syllabus for course 2.1.2">
<meta name="generator" content="bookdown 0.36 with bs4_book()">
<meta property="og:title" content=" Introduction | 2.1.2 Genomics &amp; Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="Course Syllabus for course 2.1.2">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content=" Introduction | 2.1.2 Genomics &amp; Transcriptomics">
<meta name="twitter:description" content="Course Syllabus for course 2.1.2">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">2.1.2 Genomics &amp; Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="active" href="index.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="" href="from-reads-to-counts.html"><span class="header-section-number">2</span> From Reads to Counts</a></li>
<li><a class="" href="EDA.html"><span class="header-section-number">3</span> Exploratory Data Analysis</a></li>
<li><a class="" href="chapter-4.html"><span class="header-section-number">4</span> Discovering Differentialy Expressed Genes (DEGs)</a></li>
<li><a class="" href="chapter-5.html"><span class="header-section-number">5</span> Data Analysis and Visualization</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="datasets.html"><span class="header-section-number">A</span> Discovering RNA-Seq Data Sets</a></li>
<li><a class="" href="a2-data_loading.html"><span class="header-section-number">B</span> Batch Loading Expression Data in R</a></li>
<li><a class="" href="a2-annotation.html"><span class="header-section-number">C</span> Annotating an RNA-Seq Experiment</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/rstudio/bookdown-demo">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><!--bookdown:title:end--><!--bookdown:title:start--><div id="chapter-1" class="section level1" number="1">
<h1>
<span class="header-section-number"></span> Introduction<a class="anchor" aria-label="anchor" href="#chapter-1"><i class="fas fa-link"></i></a>
</h1>
<p>Course 2.1.2 involves two separate projects performed on a single, published, data set. First, we start by performing a <em>Genomics</em> analysis similar to the case study from module 2.1.1. Then, we take the result of the mapping step, process this into read counts and perform a subsequent <em>Transcriptomics</em> analysis.</p>
<div id="genomics" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> 2.1.2 Genomics<a class="anchor" aria-label="anchor" href="#genomics"><i class="fas fa-link"></i></a>
</h2>
<p>For this Genomics analysis we are going to look at variants between the chosen data and a reference genome. One important aspect during the analysis is often assessing the quality, both from the input data set as well as the output of certain steps. Once a set of variants have been determined they can be annotated and finally linked to the biology underlying the original experiment, either reproducing their findings or aiming for novel ideas.</p>
<div id="project-deliverables" class="section level3" number="1.1.1">
<h3>
<span class="header-section-number">1.1.1</span> Project Deliverables<a class="anchor" aria-label="anchor" href="#project-deliverables"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>
<strong>Genomics pipeline</strong>: The genomics analysis pipeline is well documented in one or more <em>lab journals</em> and contains the correct tools based on the research question and data set.</li>
<li>
<strong>Presentation</strong>: The research results will be presented in an attractive method and proper level given the target audience.</li>
</ul>
</div>
<div id="data-processing" class="section level3" number="1.1.2">
<h3>
<span class="header-section-number">1.1.2</span> Data Processing<a class="anchor" aria-label="anchor" href="#data-processing"><i class="fas fa-link"></i></a>
</h3>
<p>Similar to the 2.1.1 NGS &amp; Mapping course we will go through similar steps to process the data (i.e. quality control, trimming, mapping, variant calling, etc.). This time however, we are not using a workflow manager such as Galaxy, but execute all the tools <em>command line</em>. Two reasons for this are that the number and size of the input files are far larger compared to the previously used data sets and the Galaxy server is not capable of handling that many large data sets. Also, having some proper terminal experience is very valuable.</p>
<div class="rmdnote">
<p><strong>Note on writing the pipeline</strong>: all steps need to be documented in RMarkdown documents. Each step must be introduced with a short description of the tool used, the version and followed by the command used to execute the tool. As this is a terminal command, we use code cells with the <code>bash</code> language. <strong>All</strong> the code cells <strong>must</strong> have the option <code>eval=FALSE</code> set. This is to prevent the code from being executed when knitting the document.</p>
<p>Furthermore, it is also required to run any command on the RStudio server provided at <a href="https://bioinf.nl/rstudio">https://bioinf.nl/rstudio</a>. This RStudio server is hosted on our <code>assemblix</code> server which is capable of both accessing the data and running tools on large data sets. The workstations we provide in the classrooms are <em>not</em> capable of these tasks. Also, this server already has a number of tools pre-installed (FastQC, trimmomatic, BWA/STAR mappers, etc.). Lastly, by running them from and RMarkdown document on the RStudio server forces you to document the commands compared to simply using <code>ssh</code>.</p>
<p><strong>Note on storing (intermediate) data</strong>: <strong>all</strong> data relevant to this project <strong>must</strong> be stored in the <code>/students/2024-2025/Thema05/</code> directory. In here, create a project directory for your team and create sub-directories as you go. Your home folder has a hard limit of 25GB of data and is therefore not sufficient for storing these project files. Use commands such as <code>chmod</code> and <code>chown</code> to allow your project partners to access the data too.</p>
<p><strong>Note on processing large amounts of data</strong>: building correct terminal commands to execute for instance a trimming step on a large number of files can be quite challenging. It is therefore <strong>required</strong> that test-data is used for <strong>all</strong> steps. Reasons for this are:</p>
<ul>
<li>Spotting errors quickly instead of having to wait several hours to see an error message</li>
<li>Inspecting resource usage (CPU, memory, disk space) to prevent overloading the system
<ul>
<li>Some steps can be very memory intensive (over 1 TB on large data sets!) and can crash the system</li>
</ul>
</li>
<li>Creates a small output set that can be used as a test set for the next step</li>
</ul>
<p><strong>Note on the use of GATK</strong>: The Genome Analysis Toolkit (GATK) is an often used set of tools for variant calling. Many publications cite the use of GATK but unfortunately do not include all needed details to reproduce the results. Due to the complexity of GATK we do not recommend its use, unless the publication is very clear on the exact workflow and settings used. Prepare to spend a lot of time on the official <a href="https://gatk.broadinstitute.org/hc/en-us">GATK website</a> browsing manuals before you can use it effectively.</p>
</div>
</div>
<div id="finding-a-data-set" class="section level3" number="1.1.3">
<h3>
<span class="header-section-number">1.1.3</span> Finding a Data Set<a class="anchor" aria-label="anchor" href="#finding-a-data-set"><i class="fas fa-link"></i></a>
</h3>
<p>Finding a suitable data set can be challenging as there are a few requirements such a data set needs to meet:</p>
<ul>
<li>It needs to have both a DNA- and RNA-Seq data set available.</li>
<li>The data set needs to be published (publication should be open access)</li>
<li>For the RNA-Seq part, there is a minimum amount of samples required
<ul>
<li>At least two sample <em>groups</em> with a minimum of 3 samples per group (<em>replicates</em>)</li>
<li>See the appendix <a href="datasets.html#datasets">chapter A1</a> for example valid experimental setups, this chapter also contains tips on what to do with it (mostly relevant for Transcriptomics).</li>
</ul>
</li>
</ul>
<p>Data is available from the NCBI <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive</a> and Gene Expression Omnibus.</p>
<p>The proper way to search for a data set suitable for both Genomics and Transcriptomics projects is by creating a query that searches for both: <a href="https://www.ncbi.nlm.nih.gov/gds/?term=(%22expression+profiling+by+high+throughput+sequencing%22%5BDataSet+Type%5D)+AND+%22genome+variation+profiling+by+high+throughput+sequencing%22%5BDataSet+Type%5D">(“expression profiling by high throughput sequencing”[DataSet Type]) AND “genome variation profiling by high throughput sequencing”[DataSet Type]</a>.</p>
<p>Note that for the genomics part, not all available samples have to be processed. Other than for transcriptomics, it is not common that there is a case/control experimental setup but rather a single condition of group with possibly a few replicates. If the data set does include many samples, it is recommended to select a subset of samples to process (do discuss the reasoning behind this selection).</p>
</div>
</div>
<div id="transcriptomics" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> 2.1.2 Transcriptomics<a class="anchor" aria-label="anchor" href="#transcriptomics"><i class="fas fa-link"></i></a>
</h2>
<p>The use of RNA-Sequencing techniques for measuring gene expression is relatively new and replaces <em>microarrays</em>, though in some cases microarrays are still used. Gene expression data gives valuable insights into the workings of cells in certain conditions. Especially when comparing for instance healthy and diseased samples it can become clear which genes are causal or under influence of a specific condition. Finding the genes of interest (genes showing differing expression accross conditions, called the Differentially Expressed Genes (DEGs)) is the goal of this project.</p>
<p>While there is no golden standard for analyzing RNA-sequencing data sets as there are many tools (all manufacturers of sequencing equipment also deliver software packages) we will use proven R libraries for processing, visualizing and analyzing publicly available data sets. This manual describes the steps often performed in a transcriptomics experiment. Use this as a very general guideline to process data accompanying the chosen published research. Note that this manual was originally written for processing <em>count</em> data and therefore does not describe the processing of NGS read files up to the count data as part of the <em>Genomics</em> analysis performed in this course. RNA-Seq counts describe the number of reads mapped to a gene which corresponds to the relative number of transcripts (mRNA sequences) of that gene present in the cell at the time of sampling.</p>
<div id="project-deliverables-1" class="section level3" number="1.2.1">
<h3>
<span class="header-section-number">1.2.1</span> Project Deliverables<a class="anchor" aria-label="anchor" href="#project-deliverables-1"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>
<strong>Lab Journal</strong>: The lab journal is readable, complete and aimed at the reproducability of the research. Relevant visualisations are include and statistical methods performed to support the biological interpretation of the results.</li>
</ul>
</div>
</div>
<div id="lab-journal" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> <em>Lab Journal</em><a class="anchor" aria-label="anchor" href="#lab-journal"><i class="fas fa-link"></i></a>
</h2>
<p>As you may know from previous projects, it is essential to keep a proper journal detailing every step you have done during the analysis. This journal is to be kept in an R markdown file, showing which steps have been taken in the analysis of the data set. This markdown should be <em>knitted</em> into a single PDF-file once the project is completed thus containing text detailing the steps and any decisions you’ve made, R-code (always visible!) and their resulting output/ images.</p>
<div class="rmdtip">
<p>
<strong>Notes</strong>: As a general advice; do not wait with
knitting this whole document until the project is done as knitting is
very prone to errors and trying to fix these in a large document is not
easy. Give each code chunk the proper attributes, including a name at
the minimum. This helps spot errors during knitting as that process
mentiones which chunk has been processed. Note that chunk-names
<em>must</em> be unique. And try to make proper use of chapters and
sections and include an (optional) table of contents.
</p>
</div>
</div>
<div id="schedule" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> Schedule<a class="anchor" aria-label="anchor" href="#schedule"><i class="fas fa-link"></i></a>
</h2>
<p>The Genomics and Transcriptomics parts both have their own schedule as listed below</p>
<div id="genomics-1" class="section level3" number="1.4.1">
<h3>
<span class="header-section-number">1.4.1</span> Genomics<a class="anchor" aria-label="anchor" href="#genomics-1"><i class="fas fa-link"></i></a>
</h3>
<p>Empty rows means the previous topic or task will most likely take more than one session.</p>
<style type="text/css">
.tg  {border-collapse:collapse;border-color:#ccc;border-spacing:0;}
.tg td{background-color:#fff;border-color:#ccc;border-style:solid;border-width:1px;color:#333;
  font-family:Arial, sans-serif;font-size:14px;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{background-color:#f0f0f0;border-color:#ccc;border-style:solid;border-width:1px;color:#333;
  font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-baqh{text-align:center;vertical-align:top}
.tg .tg-c3ow{border-color:inherit;text-align:center;vertical-align:top}
.tg .tg-0pky{border-color:inherit;text-align:left;vertical-align:top}
</style>
<div class="inline-table"><table class="tg">
<thead><tr>
<th class="tg-baqh">
Week
</th>
<th class="tg-baqh">
Lesson
</th>
<th class="tg-baqh">
Subject
</th>
</tr></thead>
<tbody>
<tr>
<td class="tg-0pky" rowspan="3">
1
</td>
<td class="tg-c3ow">
1
</td>
<td class="tg-0pky">
Finding a suitable experiment
</td>
</tr>
<tr>
<td class="tg-c3ow">
2
</td>
<td class="tg-0pky">
Statistics - recap and experimental setup
</td>
</tr>
<tr>
<td class="tg-c3ow">
3
</td>
<td class="tg-0pky">
Writing a Project Proposal
</td>
</tr>
<tr>
<td class="tg-0pky" rowspan="3">
2
</td>
<td class="tg-c3ow">
1
</td>
<td class="tg-0pky">
Presenting Project Proposal
</td>
</tr>
<tr>
<td class="tg-c3ow">
2
</td>
<td class="tg-0pky">
NGS Quality Control
</td>
</tr>
<tr>
<td class="tg-c3ow">
3
</td>
<td class="tg-0pky">
</td>
</tr>
<tr>
<td class="tg-0pky" rowspan="3">
3
</td>
<td class="tg-c3ow">
1
</td>
<td class="tg-0pky">
Read Mapping and QC
</td>
</tr>
<tr>
<td class="tg-c3ow">
2
</td>
<td class="tg-0pky">
</td>
</tr>
<tr>
<td class="tg-c3ow">
3
</td>
<td class="tg-0pky">
Variant Calling
</td>
</tr>
<tr>
<td class="tg-0pky" rowspan="4">
4
</td>
<td class="tg-c3ow">
1
</td>
<td class="tg-0pky">
Annotation
</td>
</tr>
<tr>
<td class="tg-c3ow">
2
</td>
<td class="tg-0pky">
</td>
</tr>
<tr>
<td class="tg-c3ow">
3
</td>
<td class="tg-0pky">
</td>
</tr>
<tr>
<td class="tg-c3ow">
4
</td>
<td class="tg-0pky">
</td>
</tr>
<tr>
<td class="tg-0pky" rowspan="2">
5
</td>
<td class="tg-c3ow">
1
</td>
<td class="tg-0pky">
Visualisation
</td>
</tr>
<tr>
<td class="tg-c3ow">
2
</td>
<td class="tg-0pky">
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="transcriptomics-1" class="section level3" number="1.4.2">
<h3>
<span class="header-section-number">1.4.2</span> Transcriptomics<a class="anchor" aria-label="anchor" href="#transcriptomics-1"><i class="fas fa-link"></i></a>
</h3>
<style type="text/css">
.tg  {border-collapse:collapse;border-color:#ccc;border-spacing:0;}
.tg td{background-color:#fff;border-color:#ccc;border-style:solid;border-width:1px;color:#333;
  font-family:Arial, sans-serif;font-size:14px;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{background-color:#f0f0f0;border-color:#ccc;border-style:solid;border-width:1px;color:#333;
  font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-baqh{text-align:center;vertical-align:top}
.tg .tg-0lax{text-align:left;vertical-align:top}
</style>
<div class="inline-table"><table class="tg">
<thead><tr>
<th class="tg-baqh">
Week
</th>
<th class="tg-baqh">
Lesson
</th>
<th class="tg-baqh">
Subject
</th>
</tr></thead>
<tbody>
<tr>
<td class="tg-0lax" rowspan="2">
5
</td>
<td class="tg-baqh">
1
</td>
<td class="tg-0lax">
Writing a Project Proposal
</td>
</tr>
<tr>
<td class="tg-baqh">
2
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-0lax" rowspan="4">
6
</td>
<td class="tg-baqh">
1
</td>
<td class="tg-0lax">
Presenting Project Proposal
</td>
</tr>
<tr>
<td class="tg-baqh">
2
</td>
<td class="tg-0lax">
Read Mapping and Quantification
</td>
</tr>
<tr>
<td class="tg-baqh">
3
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-baqh">
4
</td>
<td class="tg-0lax">
Statistics (distributions, normalization, PCA)
</td>
</tr>
<tr>
<td class="tg-0lax" rowspan="5">
7
</td>
<td class="tg-baqh">
1
</td>
<td class="tg-0lax">
Exploratory Data Analysis (<a href="#chapter-3">chapter 3</a>)
</td>
</tr>
<tr>
<td class="tg-baqh">
2
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-baqh">
3
</td>
<td class="tg-0lax">
Statistics (batch effects, regression, linear models)
</td>
</tr>
<tr>
<td class="tg-baqh">
4
</td>
<td class="tg-0lax">
Finding Differentially Expressed Genes (<a href="chapter-4.html#chapter-4">chapter 4</a>)
</td>
</tr>
<tr>
<td class="tg-baqh">
5
</td>
<td class="tg-0lax">
Statistics (Enrichment analysis)
</td>
</tr>
<tr>
<td class="tg-0lax" rowspan="6">
8
</td>
<td class="tg-baqh">
1
</td>
<td class="tg-0lax">
Data Analysis and Visualization (<a href="chapter-5.html#chapter-5">chapter 5</a>)
</td>
</tr>
<tr>
<td class="tg-baqh">
2
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-baqh">
3
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-baqh">
4
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-baqh">
5
</td>
<td class="tg-0lax">
</td>
</tr>
<tr>
<td class="tg-baqh">
6
</td>
<td class="tg-0lax">
Hand in Lab Journal
</td>
</tr>
</tbody>
</table></div>
</div>
</div>
<div id="literature" class="section level2" number="1.5">
<h2>
<span class="header-section-number">1.5</span> Literature<a class="anchor" aria-label="anchor" href="#literature"><i class="fas fa-link"></i></a>
</h2>
<p>The following (online) documents can be used throughout this course:</p>
<ul>
<li>Gene Set Enrichment Analysis – Theory. Available from: <a href="https://www.pathwaycommons.org/guide/primers/data_analysis/gsea/">https://www.pathwaycommons.org/guide/primers/data_analysis/gsea/</a>
</li>
<li>Li D. Statistical Methods for RNA Sequencing Data Analysis. In: Husi H, editor. Computational Biology [Internet]. Brisbane (AU): Codon Publications; 2019 Nov 21. Chapter 6. Available from: <a href="https://www.ncbi.nlm.nih.gov/books/NBK550334/">https://www.ncbi.nlm.nih.gov/books/NBK550334/</a>
</li>
<li>Yu, L., Fernandez, S. &amp; Brock, G. Power analysis for RNA-Seq differential expression studies. BMC Bioinformatics 18, 234 (2017). <a href="https://doi.org/10.1186/s12859-017-1648-2">https://doi.org/10.1186/s12859-017-1648-2</a>
</li>
<li>Sun, S., Hood, M., Scott, L., Peng, Q., Mukherjee, S., Tung, J., &amp; Zhou, X. (2017). Differential expression analysis for RNAseq using Poisson mixed models. Nucleic acids research, 45(11), e106. <a href="https://doi.org/10.1093/nar/gkx204">https://doi.org/10.1093/nar/gkx204</a>
</li>
</ul>
</div>
</div>
  <div class="chapter-nav">
<div class="empty"></div>
<div class="next"><a href="from-reads-to-counts.html"><span class="header-section-number">2</span> From Reads to Counts</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#chapter-1"><span class="header-section-number">1</span> Introduction</a></li>
<li>
<a class="nav-link" href="#genomics"><span class="header-section-number">1.1</span> 2.1.2 Genomics</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#project-deliverables"><span class="header-section-number">1.1.1</span> Project Deliverables</a></li>
<li><a class="nav-link" href="#data-processing"><span class="header-section-number">1.1.2</span> Data Processing</a></li>
<li><a class="nav-link" href="#finding-a-data-set"><span class="header-section-number">1.1.3</span> Finding a Data Set</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#transcriptomics"><span class="header-section-number">1.2</span> 2.1.2 Transcriptomics</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#project-deliverables-1"><span class="header-section-number">1.2.1</span> Project Deliverables</a></li></ul>
</li>
<li><a class="nav-link" href="#lab-journal"><span class="header-section-number">1.3</span> Lab Journal</a></li>
<li>
<a class="nav-link" href="#schedule"><span class="header-section-number">1.4</span> Schedule</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#genomics-1"><span class="header-section-number">1.4.1</span> Genomics</a></li>
<li><a class="nav-link" href="#transcriptomics-1"><span class="header-section-number">1.4.2</span> Transcriptomics</a></li>
</ul>
</li>
<li><a class="nav-link" href="#literature"><span class="header-section-number">1.5</span> Literature</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/rstudio/bookdown-demo/blob/master/index.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/rstudio/bookdown-demo/edit/master/index.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>2.1.2 Genomics &amp; Transcriptomics</strong>" was written by Marcel Kempenaar. It was last built on 2024-09-08.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
